ONSdigital/django-oauth2-test
# Pushing BDD tests created so far

Basic Python Behave tests to test tokens given out from the OAuth2 server. 

The task for this work has recently been moved to a backlog (see: [here](https://trello.com/c/IGvyFJDv/113-us009-012-write-tdd-titles-for-login-tests-with-oauth2) ), so this PR is just to upload the work carried out so far. 

**How to run the tests**
Switch to relevant virtualenv.
Start the Django Oauth2 server (locally) from the command line:
```
cd oauth2server
python manage.py runserver
```
You can check if this is up and running correctly by ensuring the following link returns a login page:  http://localhost:8000/admin/login/?next=/admin/

Next, run the following on the command line:
```
pip install -r requirements-for-tests.txt
behave tests/behave-BDD-tests/features 
```
